### PR TITLE
adds feed helpers

### PR DESCRIPTION
**Pull request by optikalefx**
Add getFeedResult and submitFeedHelper to help manage submitting and fetching feeds from selling partner

### CODE REVIEW

File: src/helpers/feed-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+export async function getFeedResult(
+  feedsApiClient: FeedsApiClientV20210630,
+  feedId: string,
+): Promise<{ hasErrors: boolean; xmlResponse: string }> {
+  let resultFeedDocumentId = ''
+  let feedStatus = ''
+  let attempts = 0
+  while (feedStatus !== 'DONE') {
+    // eslint-disable-next-line no-await-in-loop
+    await sleep(4)
```
Comment (josh-fisher) [inline src/helpers/feed-helpers.ts]: Should attempts and sleep time be configurable? Is it guaranteed to never take longer than 60 seconds?

File: src/helpers/feed-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+ * Given a feedClient and feedId, will loop 15 times x4 seconds delay to fetch the feed results
+ * We will parse the XML only to look for errors, which should help the user decide if they need to parse further
+ * @param feedsApiClient
+ * @param feedId
+ * @returns { hasErrors: boolean, xmlResponse: string }
+ */
+export async function getFeedResult(
+  feedsApiClient: FeedsApiClientV20210630,
+  feedId: string,
+): Promise<{ hasErrors: boolean; xmlResponse: string }> {
```
Comment (josh-fisher) [inline src/helpers/feed-helpers.ts]: Should really bring this out to it's own interface for clarity

File: src/helpers/feed-helpers.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  return new Promise<void>((resolve) => {
+    setTimeout(() => {
+      resolve()
+    }, secs * 1000)
+  })
+}
+
+/**
+ * Submit a feed to Amazon Selling Partner
+ * @param feedsApiClient
```
Comment (josh-fisher) [inline src/helpers/feed-helpers.ts]: Why have JSDoc params here if they are not given a full description and type? Same for @returns

File: src/helpers/feed-helpers.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+// eslint-disable-next-line import/no-cycle
+import { FeedsApiClientV20210630 } from '../api-clients/feeds-api-client-v20210630'
+
+/**
+ * Sleep helper
+ * @param secs
+ * @returns
+ */
+function sleep(secs: number) {
```
Comment (josh-fisher) [inline src/helpers/feed-helpers.ts]: Probably should be part of another helper module not under feed-helpers.

File: package.json [hunk 5]
```diff
@@ -1,97 +1,97 @@
...
+	"scripts": {
+		"prebuild": "npm run clean",
+		"build": "tsc --build tsconfig.build.json",
+		"clean": "rimraf lib/*",
+		"dev": "ts-node-dev --respawn --transpileOnly src",
+		"docs": "typedoc",
+		"generate-models": "ts-node utils/generate",
+		"lint": "eslint --ext ts,js src/ test/",
+		"lint:fix": "npm run lint -- --fix",
+		"semantic-release": "npx @scaleleap/semantic-release-config",
```
Comment (josh-fisher) [inline package.json]: This can be removed.

File: package.json [hunk 6]
```diff
@@ -1,97 +1,97 @@
...
+			"name": "Roman Filippov",
+			"email": "roman@scaleleap.com",
+			"url": "https://www.scaleleap.com/"
+		},
+		{
+			"name": "Toan Nguyen",
+			"url": "https://github.com/nguyentoanit"
+		}
+	],
+	"homepage": "https://github.com/ScaleLeap/selling-partner-api-sdk",
```
Comment (josh-fisher) [inline package.json]: Should we change these to whitebox-co repo?

**Comment by josh-fisher** [comment]
Base branch should be `develop`. It doesnt look like I ever merged any of my changes to `master`. Additionally, npm publish was ran manually as we never setup any thing with semantic release as this was simply a fork and was supposed to be a temporary forked solution until certain features went public :)

File: src/helpers/feed-helpers.ts (hunk 1 above)
Comment (optikalefx) [reply src/helpers/feed-helpers.ts]: Good call. I'll just make these configurable with defaults. It's not always going to be those.

**Comment by optikalefx** [comment]
> Base branch should be `develop`. It doesnt look like I ever merged any of my changes to `master`. Additionally, npm publish was ran manually as we never setup any thing with semantic release as this was simply a fork and was supposed to be a temporary forked solution until certain features went public :)

So what should we do about our release version? should we publish 1.0 or should we publish 5.1?

**Comment by josh-fisher** [comment]
> > Base branch should be `develop`. It doesnt look like I ever merged any of my changes to `master`. Additionally, npm publish was ran manually as we never setup any thing with semantic release as this was simply a fork and was supposed to be a temporary forked solution until certain features went public :)
> 
> So what should we do about our release version? should we publish 1.0 or should we publish 5.1?

Our current published @whitebox-co version is 1.0.0. This new one would likely be 1.1.0 and need to be manually published to our @whitebox-co npm manually. Probably should just go ahead and change everything in the package to @whitebox-co specifics and version to reflect our version in anticipation of moving this to our own repo instead of a fork.

**Comment by josh-fisher** [comment]
We are also over 100+ commits out of sync with the scale leap repo. Mainly dep and dev-dep updates but there are a few generation updates as well. When we move over to our own we should first merge in those changes.

**Pull request merged**